src/main/java/reposense/parser/ZoneIdArgumentType.java

```diff
@@ -0,0 +1,26 @@
+package reposense.parser;
+
+import java.time.DateTimeException;
+import java.time.ZoneId;
+
+import net.sourceforge.argparse4j.inf.Argument;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.ArgumentType;
+
+/**
+ * Verifies and parses a string-formatted zone id to a {@code ZoneId} object.
+ */
+public class ZoneIdArgumentType implements ArgumentType<ZoneId> {
```

Comment:
If you move `private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("E MMM d HH:mm:ssXXX yyyy");` into this class, you can expose the argument as a String, `ArgumentType<String>`.